src/streamlink/stream/dash.py

```diff
@@ -234,6 +234,8 @@ def parse_manifest(cls, session, url, **args):
         return ret
 
     def open(self):
+        headers = self.args.pop("headers", {})
+        self.session.http.headers.update(headers)
```

Comment:
Rather than changing the headers in the global session in might be better to use 
```python
request_args = copy.deepcopy(self.reader.stream.args)
headers = request_args.pop("headers", {})
...
return self.session.http.get(segment.url,
                             timeout=self.timeout,
                             exception=StreamError,
                             headers=headers,
                             **request_args)
``` 
in place of 
https://github.com/streamlink/streamlink/blob/0269092c2eec463817e6ac6e6f798cbffae171cd/src/streamlink/stream/dash.py#L32 
and
```python
res = self.session.http.get(self.mpd.url, exception=StreamError, **self.stream.args)
```
in place of 
https://github.com/streamlink/streamlink/blob/0269092c2eec463817e6ac6e6f798cbffae171cd/src/streamlink/stream/dash.py#L111

To maintain more consistency with the call from `parse_manifest`.